packages/block-editor/src/components/block-list/block.js

```diff
@@ -329,6 +330,24 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, registry ) => {
 				}
 			}
 
+			/**
+			 * Return true if the block content is empty.
+			 * @param {string} blockClientId The block client ID.
+			 * @return {boolean} Whether the block content is empty.
+			 */
+			function isBlockContentEmpty( blockClientId ) {
```

Comment:
Why would we need this function instead of just using `isUnmodifiedBlock`?

I was curious if the fix could be simpler and opened [a draft PR](https://github.com/WordPress/gutenberg/pull/65266) to check the CI..

Reply:
This is to handle some blocks that are modified but still considered "empty". For instance, a heading block with some attributes set but without the content. There's also an added test case in this PR.

I'm not entirely sure if this is the intended way though. We can probably define another "rule" to fix the same thing in https://github.com/WordPress/gutenberg/issues/65174.

Reply:
Okay, got it. Thank you.